cmd/monitor.go

```diff
@@ -0,0 +1,242 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/VirusTotal/vt-cli/utils"
+	vt "github.com/VirusTotal/vt-go"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var base64RegExp = `^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?$`
+
+// NewMonitorItemsListCmd returns a list or monitor_items according to a filter.
+func NewMonitorItemsListCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List monitor items in your account",
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			p, err := NewPrinter(cmd)
+			if err != nil {
+				return err
+			}
+			return p.PrintCollection(vt.URL("monitor/items"))
+		},
+	}
+
+	addIncludeExcludeFlags(cmd.Flags())
+	addFilterFlag(cmd.Flags(), "path:/")
+	addLimitFlag(cmd.Flags())
+	addCursorFlag(cmd.Flags())
+
+	return cmd
+}
+
+// Upload files to Monitor
+
+type monitorFileUpload struct {
+	scanner *vt.FileScanner
+}
+
+type uploadParams struct {
+	filePath   string
+	remotePath string
+}
+
+func (s *monitorFileUpload) Do(file interface{}, ds *utils.DoerState) string {
+	params := file.(uploadParams)
+
+	progressCh := make(chan float32)
+	defer close(progressCh)
+
+	go func() {
+		for progress := range progressCh {
+			if progress < 100 {
+				ds.Progress = fmt.Sprintf("%s uploading... %4.1f%%", params.filePath, progress)
+			} else {
+				ds.Progress = fmt.Sprintf("%s done.", params.filePath)
+			}
+		}
+	}()
+
+	f, err := os.Open(params.filePath)
+	if err != nil {
+		return fmt.Sprintf("%s", err)
+	}
+	defer f.Close()
+
+	item, err := s.scanner.MonitorUploadFilename(f, params.remotePath, progressCh)
+	if err != nil {
+		return fmt.Sprintf("%s", err)
+	}
+
+	return fmt.Sprintf("%s %s", params.filePath, item.ID())
+}
+
+var monitorItemUploadCmdHelp = `Upload a file or files contained in a folder.
+
+This command receives one file or folder path and uploads them to your
+VirusTotal Monitor account. It returns uploaded the file paths followed by their
+corresponding monitor ID.
+You can use the "vt monitor items [monitor_id]" command for retrieving
+information about the it.`
+
+var monitorItemUploadCmdExample = `  vt monitor item upload foo.exe /remote_folder/foo.exe
+  vt monitor item upload myfolder/ /another_remote_folder/`
+
+// NewMonitorItemsUploadCmd returns a new instance of the 'mointor upload file' command.
+func NewMonitorItemsUploadCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "upload [file/folder] [remote_path]",
+		Short:   "Upload one or more files to your account",
+		Long:    monitorItemUploadCmdHelp,
+		Example: monitorItemUploadCmdExample,
+		Args:    cobra.ExactArgs(2),
+
+		RunE: func(cmd *cobra.Command, args []string) error {
```

Comment:
This function is long enough to be put outside of `NewMonitorItemsCmd`.

Reply:
resolved in 87b58c2bb762313b6239b620663a58ba321ebc35